pandas/core/internals.py

```diff
@@ -4802,6 +4802,11 @@ def get_reindexed_values(self, empty_dtype, upcasted_na):
 
             if self.is_null and not getattr(self.block, 'is_categorical',
                                             None):
+                if isinstance(empty_dtype, DatetimeTZDtype):
+                    # handle timezone-aware all NaT cases
```

Comment:
what i mean is make this an else-if so this cleanly handles the different dtypes (categorical, datetime with tz) and is obvious what is going on